### DISCUSSION TITLE
Libaries needed in arduino IDE

### DISCUSSION THREAD

**Discussion by bwired-nl** [Jeroen88/EasyOpenTherm#10] [Q&A] [answered]
Hi Jeroen
can  you please add the exact list of libraries needed to compile, so which ones to install in the arduino IDE

so the libs from this list

#include <Arduino.h>

#include <WiFi.h>
#include <WiFiClientSecure.h>
#include <time.h>

#include <Wire.h>
#include <Adafruit_Sensor.h>
#include <Adafruit_BME280.h>

**Comment by Jeroen88** [comment, marked as answer]
Hi Pieter, I do not understand your question exactly? The obvious answer is: all those libraries are needed.

Maybe this is what you are looking for:
- If you compile using the Arduino IDE you can omit <Arduino.h> (but it will be added by the Arduino IDE itself). I believe that if you use PlatformIO it needs to be included explicitly
- You may omit <WiFi.h> but it will be included by <WiFiClientSecure.h> (or WiFiClient.h> if you adapted the program for this)
- You may omit <time.h> if you do not use a <WiFiClientSecure.h> TLS connection. In this case you must remove the setting of the time zone
- You may remove <Wire.h>, <Adafruit_sensor.h> and <Adafruit_BME280.h> if you use another way to inform the thermostat of the temperature. Options are: another sensor, e.g. a Dallas DS18B20 (but then you have to include the right libraries for that sensor) or get it by subscribing to a MQTT topic that publishes the room temperature on a regular bases

Are these the answers you are looking for?

Please also see the updated information about the [new MQTT thermostat](https://github.com/Jeroen88/EasyOpenTherm/issues/8#issuecomment-1321084419) that will be published soon!
